src/main/java/edu/cornell/kfs/fp/document/validation/impl/CuDisbursementVoucherDocumentPreRules.java

```diff
@@ -1,67 +1,242 @@
+   * @return Returns true if the user wishes to correct the Check Stub Text, false otherwise.
+   */
+  @SuppressWarnings("deprecation")
+  protected boolean checkCheckStubTextOverflow(DisbursementVoucherDocument dvDocument) {
+    boolean checkStubTextOK = true;
+    int maxCheckStubLines = 4; // This is the number allowed in the FastTrack file format
+
+    List<PaymentNoteText> pntList = buildNotesForCheckStubText(dvDocument.getDisbVchrCheckStubText());
+
+    if(pntList.size() > maxCheckStubLines) {
+      String questionText = SpringContext.getBean(ConfigurationService.class).getPropertyValueAsString(CUKFSKeyConstants.QUESTION_OVERSIZED_CHECK_STUB_TEXT);
+      boolean useOversizedCheckStubText = super.askOrAnalyzeYesNoQuestion(CuFPConstants.USE_OVERSIZED_CHECK_STUB_TEXT_QUESTION_ID, questionText);
+
+      if(!useOversizedCheckStubText) {
+        // return to document if the user doesn't want to use the oversized Check Stub Text
+        super.event.setActionForwardName(KFSConstants.MAPPING_BASIC);
+        checkStubTextOK = false;
+      }
+
+    }
+
+    return checkStubTextOK;
+  }
+
+  /* This is borrowed mostly from
+   * edu.cornell.kfs.document.service.impl.CuDisbursementVoucherExtractionHelperServiceImpl#buildNoteForCheckStubText
+   * with the modification of removing the setting for the previous line count (as we don't have previous lines, and
+   * we're really only interested in how the existing Check Stub Text lines will be broken up).
+   */
+  private List<PaymentNoteText> buildNotesForCheckStubText(String text) {
+    PaymentNoteText pnt = null;
+    List<PaymentNoteText> pnts = new ArrayList<PaymentNoteText>();
+    String maxNoteLinesParam = SpringContext.getBean(ParameterService.class)
+                                            .getParameterValueAsString(KfsParameterConstants.PRE_DISBURSEMENT_ALL.class, 
+                                                                       PdpParameterConstants.MAX_NOTE_LINES);
+    int previousLineCount = 0;
+    int maxNoteLines = 3;
+    try {
+      if(StringUtils.isNotBlank(maxNoteLinesParam)) {
+        maxNoteLines = Integer.parseInt(maxNoteLinesParam);
+      }
+    } catch (NumberFormatException nfe) {
+      throw new IllegalArgumentException("Invalid Max Notes Lines parameter, value: "+maxNoteLinesParam+" cannot be converted to an integer");
+    }
+
+    // The WordUtils should be sufficient for the majority of cases.  This method will
+    // word wrap the whole string based on the MAX_NOTE_LINE_SIZE, separating each wrapped
+    // word by a newline character.  The 'wrap' method adds line feeds to the end causing
+    // the character length to exceed the max length by 1, hence the need for the replace
+    // method before splitting.
+    String   wrappedText = WordUtils.wrap(text, CuDisbursementVoucherConstants.DV_EXTRACT_MAX_NOTE_LINE_SIZE);
+    String[] noteLines   = wrappedText.replaceAll("[\r]", "").split("\\n");
```

Comment:
Has the NullPointerException problem mentioned in Cathy's user story comments been resolved by these changes? I think this particular line was the one causing issues before, and it looks like it could still cause problems if the "text" parameter is null (since WordUtils.wrap() will return null if the input is null).